pkg/image/api/helper.go

```diff
@@ -157,12 +157,28 @@ func (r DockerImageReference) String() string {
 		registry += "/"
 	}
 
-	if len(r.Namespace) == 0 {
-		r.Namespace = DockerDefaultNamespace
+	namespace := r.Namespace
+	if len(namespace) == 0 {
+		namespace = DockerDefaultNamespace
+	}
+	namespace += "/"
+
+	return fmt.Sprintf("%s%s%s", registry, namespace, r.NameString())
+}
+
+// StringNoTag converts a DockerImageReference to a Docker pull spec with no tag.
+func (r DockerImageReference) StringNoTag() string {
```

Comment:
This should be `NoTag` on the Reference, and then the user should print the tag.  We should never manipulate the reference during String generation.